### PR TITLE
Convert ReactPropAnnotationSetterSpecTest to Kotlin

### PR DESCRIPTION
**Pull request by fabioh8010**
## Summary:

This PR converts `ReactPropAnnotationSetterSpecTest.java` to Kotlin as requested in [this issue](https://github.com/facebook/react-native/issues/38825).

## Changelog:

[INTERNAL] [CHANGED] - Convert ReactPropAnnotationSetterSpecTest to Kotlin

## Test Plan:

1. Run `./gradlew :packages:react-native:ReactAndroid:test`.
1. All tests should pass.

### CODE REVIEW

**Comment by fabioh8010** [comment]
@cortinico Could you please review this one? Thanks!

I did two things besides converting the test to Kotlin:
* Fixed some typos
* To make the test build and work, changed return of `createShadowNodeInstance()` and `createViewInstance()`

**Comment by fabioh8010** [comment]
Tests failed because of this:

```
> Task :packages:react-native:ReactAndroid:compileDebugUnitTestKotlin FAILED
e: warnings found and -Werror specified
w: file:///root/react-native/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:44:47 Parameter 'v' is never used
w: file:///root/react-native/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:44:57 Parameter 'value' is never used
w: file:///root/react-native/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:44:73 Parameter 'otherValue' is never used
w: file:///root/react-native/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:52:64 Parameter 'v' is never used
w: file:///root/react-native/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:60:72 Parameter 'v' is never used
w: file:///root/react-native/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:60:82 Parameter 'value' is never used
w: file:///root/react-native/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:68:64 Parameter 'v' is never used
w: file:///root/react-native/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:68:73 Parameter 'value' is never used
w: file:///root/react-native/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:77:47 Parameter 'v' is never used
w: file:///root/react-native/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:77:57 Parameter 'index' is never used
w: file:///root/react-native/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:77:69 Parameter 'value' is never used
w: file:///root/react-native/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:77:83 Parameter 'other' is never used
w: file:///root/react-native/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:86:38 Parameter 'v' is never used
w: file:///root/react-native/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:86:48 Parameter 'index' is never used
w: file:///root/react-native/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:95:39 Parameter 'v' is never used
w: file:///root/react-native/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:95:49 Parameter 'value' is never used
w: file:///root/react-native/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:95:63 Parameter 'sth' is never used
w: file:///root/react-native/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:104:38 Parameter 'v' is never used
w: file:///root/react-native/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:104:47 Parameter 'index' is never used
w: file:///root/react-native/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:104:59 Parameter 'value' is never used
w: file:///root/react-native/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:113:45 Parameter 'v' is never used
w: file:///root/react-native/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:113:55 Parameter 'index' is never used
w: file:///root/react-native/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:113:67 Parameter 'value' is never used

> Task :packages:react-native:ReactAndroid:compileReleaseUnitTestKotlin FAILED
e: warnings found and -Werror specified
w: file:///root/react-native/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:44:47 Parameter 'v' is never used
w: file:///root/react-native/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:44:57 Parameter 'value' is never used
w: file:///root/react-native/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:44:73 Parameter 'otherValue' is never used
w: file:///root/react-native/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:52:64 Parameter 'v' is never used
w: file:///root/react-native/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:60:72 Parameter 'v' is never used
w: file:///root/react-native/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:60:82 Parameter 'value' is never used
w: file:///root/react-native/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:68:64 Parameter 'v' is never used
w: file:///root/react-native/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:68:73 Parameter 'value' is never used
w: file:///root/react-native/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:77:47 Parameter 'v' is never used
w: file:///root/react-native/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:77:57 Parameter 'index' is never used
w: file:///root/react-native/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:77:69 Parameter 'value' is never used
w: file:///root/react-native/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:77:83 Parameter 'other' is never used
w: file:///root/react-native/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:86:38 Parameter 'v' is never used
w: file:///root/react-native/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:86:48 Parameter 'index' is never used
w: file:///root/react-native/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:95:39 Parameter 'v' is never used
w: file:///root/react-native/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:95:49 Parameter 'value' is never used
w: file:///root/react-native/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:95:63 Parameter 'sth' is never used
w: file:///root/react-native/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:104:38 Parameter 'v' is never used
w: file:///root/react-native/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:104:47 Parameter 'index' is never used
w: file:///root/react-native/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:104:59 Parameter 'value' is never used
w: file:///root/react-native/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:113:45 Parameter 'v' is never used
w: file:///root/react-native/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:113:55 Parameter 'index' is never used
w: file:///root/react-native/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:113:67 Parameter 'value' is never used
```

We have some warnings related to unused parameters but we need them in the code.

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+/** Test that verifies that spec of methods annotated with @ReactProp is correct */
+@RunWith(RobolectricTestRunner::class)
+class ReactPropAnnotationSetterSpecTest {
+  private abstract inner class BaseViewManager : ViewManager<View, ReactShadowNode<*>>() {
+    override fun getName(): String {
+      return "IgnoredName"
+    }
```
Comment (cortinico) [inline packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:23]:
```suggestion
    override fun getName(): String = "IgnoredName"
```

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+    override fun createShadowNodeInstance(): ReactShadowNode<*> {
+      return createShadowNodeInstance()
+    }
```
Comment (cortinico) [inline packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:27]:
```suggestion
    override fun createShadowNodeInstance(): ReactShadowNode<*> = createShadowNodeInstance()
```

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+    override fun getShadowNodeClass(): Class<out ReactShadowNode<*>> {
+      return ReactShadowNode::class.java
```
Comment (cortinico) [inline packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:30]: same here, use expression return

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    }
+
+    override fun createViewInstance(reactContext: ThemedReactContext): View {
+      return createViewInstance(reactContext)
```
Comment (cortinico) [inline packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:34]: same here, use expression return

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+import com.facebook.react.uimanager.annotations.ReactProp
+import com.facebook.react.uimanager.annotations.ReactPropGroup
+import java.util.Date
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+/** Test that verifies that spec of methods annotated with @ReactProp is correct */
+@RunWith(RobolectricTestRunner::class)
+class ReactPropAnnotationSetterSpecTest {
```
Comment (cortinico) [inline packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropAnnotationSetterSpecTest.kt:19]: Please add a `@Suppress("UNUSED_PARAMETER")` on this class to suppress the warning which you correctly need

**Review by cortinico — CHANGES_REQUESTED** [review]
Thanks for sending this over @fabioh8010 
Let's handle the comments I left and we can merge this

**Comment by fabioh8010** [comment]
@cortinico I've addressed the changes, sorry for the late!

**Pull request closed without merging**